### PR TITLE
Add diagnostics, and make WPI scripts more consistent

### PR DESCRIPTION
**Pull request by mernst**

### CODE REVIEW

File: checker/bin/wpi-many.sh [hunk 1]
```diff
@@ -85,8 +85,8 @@ if [ "${has_java_home}" = "yes" ] && [ ! -d "${JAVA_HOME}" ]; then
     exit 1
 fi
 
+java_version=$("${JAVA_HOME}"/bin/java -version 2>&1 | head -1 | cut -d'"' -f2 | sed '/^1\./s///' | cut -d'.' -f1)
 if [ "${has_java_home}" = "yes" ]; then
-    java_version=$("${JAVA_HOME}"/bin/java -version 2>&1 | head -1 | cut -d'"' -f2 | sed '/^1\./s///' | cut -d'.' -f1)
```
Comment (kelloggm) [inline checker/bin/wpi-many.sh:89]: If this is moved outside the test of `${has_java_home}`, then won't this command always fail when the `$JAVA_HOME` environment variable isn't specified? That seems undesirable, but maybe I just don't understand the motivation for this change.

File: checker/bin/wpi.sh [hunk 2]
```diff
@@ -33,9 +33,9 @@ SCRIPTDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
 SCRIPTPATH="${SCRIPTDIR}/wpi.sh"
 
 # Report line numbers when the script fails, from
-# https://unix.stackexchange.com/a/522815
+# https://unix.stackexchange.com/a/522815 .
```
Comment (kelloggm) [inline checker/bin/wpi.sh:36]: What is the purpose of this change?

File: checker/bin/wpi.sh [hunk 3]
```diff
@@ -75,8 +75,13 @@ else
   has_java19="yes"
 fi
 
+if [ "${has_java_home}" = "yes" ] && [ ! -d "${JAVA_HOME}" ]; then
+    echo "JAVA_HOME is set to a non-existent directory ${JAVA_HOME}"
+    exit 1
+fi
+
+java_version=$("${JAVA_HOME}"/bin/java -version 2>&1 | head -1 | cut -d'"' -f2 | sed '/^1\./s///' | cut -d'.' -f1)
```
Comment (kelloggm) [inline checker/bin/wpi.sh:83]: Same concern as above.

**Review by kelloggm — CHANGES_REQUESTED** [review]

File: checker/bin/wpi-many.sh (hunk 1 above)
Comment (mernst) [reply checker/bin/wpi-many.sh]: Good point.  I reverted this change.

File: checker/bin/wpi.sh (hunk 2 above)
Comment (mernst) [reply checker/bin/wpi.sh:36]: That period appears in wpi-many.sh but not in this file.  I made the files more consistent, so that their diffs are smaller and it's easier to compare them and to copy changes from one to the other.

File: checker/bin/wpi.sh (hunk 3 above)
Comment (mernst) [reply checker/bin/wpi.sh]: I reverted this change.

**Comment by mernst** [comment]
@kelloggm That failure was a nondeterministic (but common) problem with CI inference jobs.
